internal/manifests/gateway.go

```diff
@@ -0,0 +1,266 @@
+package manifests
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"path"
+
+	"github.com/ViaQ/logerr/kverrors"
+	"github.com/imdario/mergo"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ViaQ/loki-operator/internal/manifests/internal/gateway"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+// BuildLokiStackGateway returns a list of k8s objects for Loki Stack Gateway
+func BuildLokiStackGateway(opts Options) ([]client.Object, string, error) {
+	gatewayConfigMap, sha1C, err := GatewayConfigMap(opts)
+	if err != nil {
+		return nil, "", err
+	}
+
+	deployment := NewLokiStackGatewayDeployment(opts)
```

Comment:
The SHA1 needs to be passed to the Deployment here to pass it down to `a := commonAnnotations(opts.ConfigSHA1)` instead of the `opts.ConfigSHA1`.